glusterd2/commands/peers/peer-rpc-svc.go

```diff
@@ -53,13 +53,22 @@ func (p *PeerService) Join(ctx context.Context, req *JoinReq) (*JoinRsp, error)
 
 	// TODO: Ensure no other operations are happening
 
-	peers, _ := peer.GetPeersF()
+	peers, err := peer.GetPeersF()
+	if err != nil {
+		logger.WithError(err).Error("failed to connect to store")
```

Comment:
@Madhu-1  Are you sure GetPeersF will return an error only if it fails to connect to store? unmarshalling error is also possible

Reply:
@rishubhjain we are not returning an error if [JSON unmarshal fails](https://github.com/gluster/glusterd2/blob/master/glusterd2/peer/store-utils.go#L133)